### PR TITLE
Enable differentiated error messages for DR eligibility

### PR DESCRIPTION
**Pull request by elireisman**
### Description

Addresses cases when DR eligibility checks fail on the API side, but the DR Action was not able to differentiate and supply specific instruction to end users.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -53,6 +53,10 @@ async function run(): Promise<void> {
     }
   } catch (error) {
     if (error instanceof RequestError && error.status === 404) {
+      core.setFailed(
+        `Dependency review could not obtain dependency data for the specified owner, repository, or revision range.`
```
[the remark this answers is not in the document]
Comment (elireisman) [reply src/main.ts:57]: I think this is the goal of the parent Issue, to differentiate between explicitly not-DR-eligible repos (403) and input problems to the API (bad repo/owner, bad commit/rev range, etc.) (404) which I don't expect to happen much when the inputs come from PRs. My estimate, you'll see those 404s drop off to nothing for DR Action callers, and the 403's will increase. Does that make sense?

OTOH - the messages themselves may or may not be what we want to relate to end users. Opinions about what 404 (I can't find the repo/owner or those revs don't exist) vs. 403 (this repo shouldn't have the DR Action installed unless you make it eligible for DR!)? cc @febuiles @courtneycl @lseppala

[the remark this answers is not in the document]
Comment (febuiles) [reply src/main.ts:57]: @elireisman Apologies, I thought the error message for DG enablement had been lost but now I see it's been moved to the 403 case!

**Review by febuiles — APPROVED** [review]

**Comment by elireisman** [comment]
Done - and I just bumped the `v1` tag to latest. FYI - this might pick up more commits than just my own ❗ it looks like it was still pinned to the original init commit for this repo. I'm watching the API graphs today 🤞 should be fine?

**Comment by febuiles** [comment]
@elireisman `v1` was used for launch, but going forward we plan to use semver for this Action. Next release should be 1.0.1. Changes on v1 should still be picked up I think.

**Pull request merged**